dataprofiler/profilers/profile_builder.py

```diff
@@ -639,57 +661,108 @@ def tqdm(l):
                 sample_ids[0] = sample_ids[0][:sample_size]
             # Sort the sample_ids anre replace prior
             sample_ids[0] = sorted(sample_ids[0])
-        
-        pool = None
-        if options.structured_options.multiprocess.is_enabled:
-            cpu_count = 1
-            try:
-                cpu_count = mp.cpu_count()
-            except NotImplementedError as e:
-                cpu_count = 1
-
-            # No additional advantage beyond 4 processes
-            # Always leave 1 cores free
-            # https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method            
-            if cpu_count > 2:
-                cpu_count = min(cpu_count-1, 4)
-                try: 
-                    pool = mp.Pool(cpu_count)
-                    print("Utilizing",cpu_count, "processes for profiling")
-                except Exception as e:
-                    pool = None
-                    warnings.warn(
-                        'Multiprocessing disabled, please change the multiprocessing'+
-                        ' start method, via: multiprocessing.set_start_method(<method>)'+
-                        ' Possible methods include: fork, spawn, forkserver, None'
-                    )
-        
-        for col in tqdm(df.columns):
-            if col in profile:
-                column_profile = profile[col]
-                column_profile.update_profile(
-                    df[col],
-                    sample_size=sample_size,
-                    min_true_samples=min_true_samples,
-                    sample_ids=sample_ids,
-                    pool=pool
-                )
-            else:
+        # Numpy arrays allocate to heap and can be shared between processes
+        # Non-locking multiprocessing fails on machines without POSIX (windows)
+        # The function handles that situation, but will be single process
+        # Newly introduced features (python3.8) improves the situation
+        sample_ids = np.array(sample_ids)        
+
+        # Create structured profile objects
+        new_cols = set()
+        for col in df.columns:
```

Comment:
Are we not using tqdm anymore? If not we should remove it from requirements and/or the "tqdm" function defined above.

Reply:
We are, but this is just creating them it's very very fast. tqdm is further below, when processing occurs.